### ISSUE TITLE
Empty HTTP 200 response on a SourceFormatException

### ISSUE THREAD

**Issue by Lucidiot** [cantaloupe-project/cantaloupe#572]
I have a Cantaloupe server in a Docker container that runs with an S3Source and S3Cache, and all the default processor settings, and with TurboJPEG available. When a JPEG image uses a `.png` extension and some transformations are requested (it's not just the full, original image), the automatic selection uses Java2D, which seems to expect a PNG file. Since that isn't the case, there is a `SourceFormatException`, but the server still replies with HTTP 200, with an empty body.

```
13:02:06.580 [qtp361571968-17] INFO  e.i.l.c.r.i.v.ImageResource - Handling GET /iiif/2/3z6t7pSN45UYit2nwcJaTPon.png/full/10,/0/default.jpg
13:02:06.580 [qtp361571968-17] DEBUG e.i.l.c.r.i.v.ImageResource - Request headers: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8; User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/99.0; X-Forwarded-Host: ark.localhost; X-Forwarded-Proto: https; Sec-Fetch-Dest: document; Sec-Fetch-Site: none; Host: ark.localhost; Accept-Encoding: gzip, deflate, br; Dnt: 1; X-Forwarded-Port: 443; Sec-Fetch-Mode: navigate; TE: trailers; Upgrade-Insecure-Requests: 1; X-Forwarded-For: 172.18.0.1; Sec-Fetch-User: ?1; Accept-Language: en-US,en;q=0.8,fr;q=0.5,fr-FR;q=0.3; X-Real-Ip: 172.18.0.1; X-Forwarded-Server: 85f02b39a2e1
13:02:06.581 [qtp361571968-17] DEBUG e.i.l.c.r.i.v.ImageResource - Base URI from assembled from base_uri key: https://dev.localhost
13:02:06.583 [qtp361571968-17] DEBUG e.i.l.c.d.DelegateProxyService - Instantiating a JavaDelegate
13:02:06.590 [qtp361571968-17] DEBUG e.i.l.c.i.MetaIdentifier - [Raw path component: 3z6t7pSN45UYit2nwcJaTPon.png] -> [decoded: 3z6t7pSN45UYit2nwcJaTPon.png] -> [slashes substituted: 3z6t7pSN45UYit2nwcJaTPon.png]
13:02:06.599 [qtp361571968-17] DEBUG e.i.l.c.i.Identifier - [Raw path component: 3z6t7pSN45UYit2nwcJaTPon.png] -> [decoded: 3z6t7pSN45UYit2nwcJaTPon.png] -> [slashes substituted: 3z6t7pSN45UYit2nwcJaTPon.png]
13:02:06.621 [qtp361571968-17] DEBUG e.i.l.c.p.ProcessorFactory - Java2dProcessor selected for format PNG (AutomaticSelectionStrategy offered Java2dProcessor)
13:02:06.621 [qtp361571968-17] DEBUG e.i.l.c.p.ProcessorConnector - Stream -> StreamProcessor connection between S3Source and Java2dProcessor
13:02:06.648 [qtp361571968-17] DEBUG e.i.l.c.c.S3Cache - getInfo(): read info/57527dc2786b77a087523decac1b13bd.json from bucket iiif-cache in 27 msec
13:02:06.649 [qtp361571968-17] DEBUG e.i.l.c.c.InfoService - getInfo(): retrieved info of 3z6t7pSN45UYit2nwcJaTPon.png from S3Cache in 28 msec
13:02:06.649 [work-lo-4444694] DEBUG e.i.l.c.c.S3Cache - touchAsync(): info/57527dc2786b77a087523decac1b13bd.json
13:02:06.664 [qtp361571968-17] DEBUG e.i.l.c.r.i.v.ImageResource - Base URI from assembled from base_uri key: https://dev.localhost
13:02:06.668 [qtp361571968-17] DEBUG e.i.l.c.c.S3Cache - newDerivativeImageInputStream(): bucket: iiif-cache; key: image/57527dc2786b77a087523decac1b13bd/25ea0fed971431efc82a29e305249e37.jpg
13:02:06.684 [qtp361571968-17] DEBUG e.i.l.c.r.ImageRepresentation - Writing to the response & derivative cache simultaneously
13:02:06.685 [qtp361571968-17] DEBUG e.i.l.c.s.S3Source - Requesting iiif/3z6t7pSN45UYit2nwcJaTPon.png
13:02:06.700 [qtp361571968-17] DEBUG e.i.l.c.p.c.p.PNGImageReader - Using com.sun.imageio.plugins.png.PNGImageReader
13:02:06.701 [qtp361571968-17] DEBUG e.i.l.c.p.c.j.JPEGImageWriter - ImageIO plugin preferences: com.sun.imageio.plugins.jpeg.JPEGImageWriter
13:02:06.701 [qtp361571968-17] DEBUG e.i.l.c.p.c.j.JPEGImageWriter - Using com.sun.imageio.plugins.jpeg.JPEGImageWriter
13:02:06.707 [qtp361571968-17] DEBUG e.i.l.c.r.ImageRepresentation - write(): Unsupported source format
edu.illinois.library.cantaloupe.processor.SourceFormatException: Unsupported source format
	at edu.illinois.library.cantaloupe.processor.codec.AbstractIIOImageReader.handle(AbstractIIOImageReader.java:64)
	at edu.illinois.library.cantaloupe.processor.codec.AbstractIIOImageReader.read(AbstractIIOImageReader.java:408)
	at edu.illinois.library.cantaloupe.processor.Java2dProcessor.process(Java2dProcessor.java:61)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.process(ImageRepresentation.java:205)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.copyOrProcess(ImageRepresentation.java:167)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.write(ImageRepresentation.java:130)
	at edu.illinois.library.cantaloupe.resource.ImageRequestHandler.handle(ImageRequestHandler.java:405)
	at edu.illinois.library.cantaloupe.resource.iiif.v2.ImageResource.doGET(ImageResource.java:128)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.handle(HandlerServlet.java:97)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.doGet(HandlerServlet.java:35)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1434)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1349)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.base/java.lang.Thread.run(Thread.java:829)
13:02:06.707 [qtp361571968-17] DEBUG e.i.l.c.r.HandlerServlet - Responded to GET /iiif/2/3z6t7pSN45UYit2nwcJaTPon.png/full/10,/0/default.jpg with HTTP 200 in 128 msec
```

`curl -v` shows this response:

```
HTTP/2 200
access-control-allow-origin: *
content-type: image/jpeg
date: Tue, 22 Mar 2022 13:08:33 GMT
link: <https://dev.localhost/iiif/2/3z6t7pSN45UYit2nwcJaTPon.png/full/full/10,/default.jpg>;rel="canonical"
server: Jetty(9.4.34.v20201102)
vary: Accept, Accept-Charset, Accept-Encoding, Accept-Language, Origin
x-powered-by: Cantaloupe/5.0.5
content-length: 0
```

**Comment by adolski** [comment]
Thanks @Lucidiot, this was an oversight that is now fixed on the `release/5.0` branch for inclusion in 5.0.6.

**Comment by Devangptrt** [comment]
@adolski I'm facing the similar issue while accessing JPEG2000 with Grok Image compression tool v9.7.5 and cantaloupe ( tried v5.0.5 and also checked from develop branch )
```09:57:38.772 [qtp444418972-21] DEBUG e.i.l.c.r.ImageRepresentation - write(): Unsupported source format: JPEG2000
edu.illinois.library.cantaloupe.processor.SourceFormatException: Unsupported source format: JPEG2000
	at edu.illinois.library.cantaloupe.processor.GrokProcessor.process(GrokProcessor.java:406)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.process(ImageRepresentation.java:205)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.copyOrProcess(ImageRepresentation.java:167)
	at edu.illinois.library.cantaloupe.resource.ImageRepresentation.write(ImageRepresentation.java:130)
	at edu.illinois.library.cantaloupe.resource.ImageRequestHandler.handle(ImageRequestHandler.java:419)
	at edu.illinois.library.cantaloupe.resource.iiif.v2.ImageResource.doGET(ImageResource.java:136)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.handle(HandlerServlet.java:97)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.doGet(HandlerServlet.java:35)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:508)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1370)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:463)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1292)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.Server.handle(Server.java:562)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:399)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:656)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:391)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:319)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SocketChannelEndPoint$1.run(SocketChannelEndPoint.java:101)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:378)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:894)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1038)
	at java.base/java.lang.Thread.run(Thread.java:829)
09:57:38.774 [qtp444418972-21] DEBUG e.i.l.c.r.HandlerServlet - Responded to GET /iiif/2/v14.jp2/full/full/0/default.jpg with HTTP 200 in 184 msec
